src/components/user/user.service.ts

```diff
@@ -433,27 +448,42 @@ export class UserService {
       .query()
       .call(matchRequestingUser, session)
       .call(matchUserPermissions, 'User', id)
-      .call(addAllSecureProperties, ...props)
+      .call(addAllSecureProperties, ...[...props, 'roles'])
       .with([
         ...props.map(addPropertyCoalesceWithClause),
+        `
+        {
+          value: collect(distinct roles.value),
```

Comment:
Doesn't `collect` always return an array? If not maybe it could be wrapped in a `coalesce(..., [])` to prevent the need to map result to response below.